File: tests_ext/src/main/java/com/github/klee0kai/test_ext/inject/di/forest/OldForestComponent.java
package com.github.klee0kai.test_ext.inject.di.forest;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.stone.types.lifecycle.IStoneLifeCycleOwner;
import com.github.klee0kai.test.di.base_forest.ForestComponent;
import com.github.klee0kai.test_ext.inject.mowgli.animal.OldHorse;

@Component
public interface OldForestComponent extends ForestComponent {


    @Override
    OldIdentityModule identity();

    DiseasesModule diseases();


    void inject(OldHorse horse, IStoneLifeCycleOwner iStoneLifeCycleOwner);


}

File: tests_ext/src/main/java/com/github/klee0kai/test_ext/inject/di/forest/OldForestComponent.kt
package com.github.klee0kai.test_ext.inject.di.forest

import com.github.klee0kai.stone.annotations.component.Component
import com.github.klee0kai.stone.annotations.component.ExtendOf
import com.github.klee0kai.stone.lifecycle.StoneLifeCycleOwner
import com.github.klee0kai.test.di.base_forest.ForestComponent
import com.github.klee0kai.test_ext.inject.mowgli.animal.OldHorse

@Component
interface OldForestComponent : ForestComponent {

    override fun identity(): OldIdentityModule

    @ExtendOf
    fun extOf(parent: ForestComponent)

    fun diseases(): DiseasesModule?

    fun inject(horse: OldHorse?, stoneLifeCycleOwner: StoneLifeCycleOwner?)

    fun inject(horse: OldHorse?)

}
